lms/djangoapps/bulk_email/tasks.py

```diff
@@ -408,6 +408,11 @@ def _get_source_address(course_id, course_title):
         re.sub(r"[^\w.-]", '_', course_id.course),
         settings.BULK_EMAIL_DEFAULT_FROM_EMAIL
     )
+
+    # make sure it's not longer than 320 chars
+    if len(from_addr) >= 320:
```

Comment:
I can verify #11055 — I'm thinking we should just truncate after we have our translated string. This isn't the best UI, but I think it's going to be pretty rare with this address goes over that 320 character limit